jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/EditAttributesDialog.kt

```diff
@@ -0,0 +1,111 @@
+    private val project: Project,
+    private val client: SqsClient,
+    private val queue: Queue,
+    private val attributes: Map<QueueAttributeName, String>
+) : DialogWrapper(project), CoroutineScope by ApplicationThreadPoolScope("EditAttributesDialog") {
+    val view = EditAttributesPanel()
+
+    init {
+        title = message("sqs.edit.attributes")
+        setOKButtonText(message("sqs.edit.attributes.save"))
+        populateFields()
+        init()
+    }
+
+    override fun createCenterPanel(): JComponent? = view.component
+
+    override fun doValidate(): ValidationInfo? {
+        try {
+            view.messageSize.validateContent()
+            view.retentionPeriod.validateContent()
+        } catch (e: ConfigurationException) {
+            return ValidationInfo(e.title)
+        }
+        return view.waitTime.validate() ?: view.deliveryDelay.validate() ?: view.waitTime.validate()
+    }
+
+    override fun doCancelAction() {
+        SqsTelemetry.editQueueAttributes(project, Result.Cancelled, queue.telemetryType())
+        super.doCancelAction()
+    }
+
+    override fun doOKAction() {
+        if (!isOKActionEnabled) {
+            return
+        }
+        isOKActionEnabled = false
+        launch {
+            try {
+                updateAttributes()
+                notifyInfo(
+                    project = project,
+                    title = message("sqs.service_name"),
+                    content = message("sqs.edit.attributes.updated", queue.queueName)
+                )
+                SqsTelemetry.editQueueAttributes(project, Result.Succeeded, queue.telemetryType())
+                withContext(getCoroutineUiContext(ModalityState.any())) {
+                    close(OK_EXIT_CODE)
+                }
+            } catch (e: SqsException) {
+                LOG.error(e) { "Updating queue attributes failed" }
+                setErrorText(e.message)
+                isOKActionEnabled = true
+                SqsTelemetry.editQueueAttributes(project, Result.Failed, queue.telemetryType())
+            }
+        }
+    }
+
+    private fun populateFields() {
+        view.visibilityTimeout.value = attributes[QueueAttributeName.VISIBILITY_TIMEOUT]?.toIntOrNull() ?: MIN_VISIBILITY_TIMEOUT
+        view.messageSize.text = attributes[QueueAttributeName.MAXIMUM_MESSAGE_SIZE]
+        view.retentionPeriod.text = attributes[QueueAttributeName.MESSAGE_RETENTION_PERIOD]
+        view.deliveryDelay.value = attributes[QueueAttributeName.DELAY_SECONDS]?.toIntOrNull() ?: MIN_DELIVERY_DELAY
+        view.waitTime.value = attributes[QueueAttributeName.RECEIVE_MESSAGE_WAIT_TIME_SECONDS]?.toIntOrNull() ?: MIN_WAIT_TIME
+    }
+
+    internal fun updateAttributes() {
+        client.setQueueAttributes {
+            it.queueUrl(queue.queueUrl)
+            it.attributes(
+                mutableMapOf(
+                    Pair(QueueAttributeName.VISIBILITY_TIMEOUT, view.visibilityTimeout.value.toString()),
```

Comment:
I think in the past we've used the `to` shortcut so `x to y` instead of `Pair(x, y)`

Reply:
Yep forgot to change this one, fixed